pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -156,74 +158,33 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
     // Check if we should disconnect (there are 0 subscribers to our data)
     if(it_pub_.getNumSubscribers() == 0 && pub_->getPublisher().getNumSubscribers() == 0)
     {
+      NODELET_DEBUG("Disconnecting.");
+      pubThread_->interrupt();
+      pubThread_->join();
+      sub_.shutdown();
+
       try
       {
-        NODELET_DEBUG("Disconnecting.");
-        pubThread_->interrupt();
-        pubThread_->join();
-        sub_.shutdown();
         NODELET_DEBUG("Stopping camera capture.");
         pg_.stop();
-        /*NODELET_DEBUG("Disconnecting from camera.");
-        pg_.disconnect();*/
       }
       catch(std::runtime_error& e)
       {
         NODELET_ERROR("%s", e.what());
       }
-    }
-    else if(!sub_)     // We need to connect
-    {
-      NODELET_DEBUG("Connecting");
-      // Try connecting to the camera
-      volatile bool connected = false;
-      while(!connected && ros::ok())
-      {
-        try
-        {
-          NODELET_DEBUG("Connecting to camera.");
-          pg_.connect(); // Probably already connected from the reconfigure thread.  This will will not throw if successfully connected.
-          connected = true;
-        }
-        catch(std::runtime_error& e)
-        {
-          NODELET_ERROR("%s", e.what());
-          ros::Duration(1.0).sleep(); // sleep for one second each time
-        }
-      }
 
-      // Set the timeout for grabbing images.
-      double timeout;
-      getMTPrivateNodeHandle().param("timeout", timeout, 1.0);
       try
       {
-        NODELET_DEBUG("Setting timeout to: %f.", timeout);
-        pg_.setTimeout(timeout);
+        NODELET_DEBUG("Disconnecting from camera.");
```

Comment:
Now I also disconnect when nothing is subscribed. I can comment it out, but in principle it doesn't make any difference: it works, camera LED doesn't change.